src/rviz/image/mouse_click.h

```diff
@@ -0,0 +1,61 @@
+#ifndef RVIZ_MOUSE_CLICK_H
+#define RVIZ_MOUSE_CLICK_H
+
+#include <QObject>
+
+#ifndef Q_MOC_RUN // See: https://bugreports.qt-project.org/browse/QTBUG-22829
+#include <iostream>
+#include <string>
+#include <boost/shared_ptr.hpp>
+
+#include <QMouseEvent>
+
+#include "ros/ros.h"
+#include "geometry_msgs/PointStamped.h"
+#include "std_msgs/String.h"
+
+#include "rviz/rviz_export.h"
+#endif
+
+
+namespace rviz
+{
+/** @brief Class for capturing mouse clicks.
+ *
+ * This class handles mouse clicking functionalities integrated into the ImageDisplay.
+ * It uses a qt event filter to capture mouse clicks, handles image resizing and checks if the click was
+ * inside or outside the image. It also scales the pixel coordinates to get them w.r.t. the image (not
+ * the window) size. Mouse clicks image pixel coordinates are published as ros geometry_msgs
+ * PointStamped. The z component is ignored. The topic name where the mouse clicks are published is
+ * defined created after the subscribed image topic as: <image_topic>/mouse_click.
+ */
+
+class RVIZ_EXPORT MouseClick : QObject
+{
+public:
+  MouseClick();
+  ~MouseClick();
+
+  void onInitialize(QObject* parent);
+
+  /** @brief ROS topic management. */
+  void publish();
+  void unpublish();
+
+  void setDimensions(int img_width, int img_height, int win_width, int win_height);
+  void setTopic(const QString& image_topic);
+  void updateTopic(const QString& image_topic);
+
+private:
+  virtual bool eventFilter(QObject* obj, QEvent* event);
+
+  bool has_dimensions_;
+  int img_width_, img_height_, win_width_, win_height_; // To assess if the clicks are inside the image.
+  boost::shared_ptr<ros::NodeHandle> node_handle_;
+  boost::shared_ptr<ros::Publisher> publisher_;
```

Comment:
Why do you need to use pointers here?

Reply:
Its been some years since I've used cpp, but at the time I was using boost pointers over regular pointers whenever possible because I thought they were safer ... not sure if that makes sense tough

Reply:
My point was to not use pointers at all for these, but regular object instances.

Reply:
Hi @rhaschke ,

So I just set the **node_handle_** as a regular object instance.

However, for the **publisher_** I think I need it to be a pointer so I can create a new publisher on a different topic in these methods:

https://github.com/miguelriemoliveira/rviz/blob/61974117adaf5f632800ccc906fe84d3204d5a93/src/rviz/image/mouse_click.cpp#L28-L40

If the **publisher_** is a regular object I don't know how to do this ...